core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -86,8 +86,14 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
-    if (!fileSystem.isDirectory(new Path(logBaseDir))) {
-      throw new IllegalArgumentException(s"Log directory $logBaseDir does not exist.")
+    try{
```

Comment:
Here's our code style guide: https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide

The Spark style would format the code like this:

```scala
try {
  if (!filesystem.getFileStatus(newPath(logBaseDir)).isDir()) {
    throw new IllegalArgumentException(s"Log directory $logBaseDir is not a directory.")
  }
} catch {
  case ioe: FileNotFoundException => fileSystem.mkdirs(newPath(logBaseDir)))
}
```